### PR TITLE
MES-1013 Vehicle Checks

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
- Record a dangerous fault against show me question
- Record a driving fault agains show me question
- Toggle show me question's completed status
- Remove show me question faults
- Update global fault counter

### CODE REVIEW

File: src/modules/tests/test-data/test-data.reducer.ts [hunk 1]
```diff
@@ -312,6 +323,39 @@ export function testDataReducer(
...
+        },
+      };
+    case testDataActions.SHOW_ME_QUESTION_REMOVE_FAULT:
+      return {
+        ...state,
+        vehicleChecks: {
+          ...state.vehicleChecks,
+          showMeQuestion: {
+            ...state.vehicleChecks.showMeQuestion,
+            outcome: undefined,
```
Comment (ammarhaiderbjss) [inline src/modules/tests/test-data/test-data.reducer.ts]: Might need to check with @ali-campbell but I think we should be removing the outcome when it is removed..

**Comment by ammarhaiderbjss** [comment]
@szabolcs-keresztes Awesome, very comprehensive and thorough tests 😍, just a query on the removing fault reducer then happy to approve.

File: src/modules/tests/test-data/test-data.reducer.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/test-data/test-data.reducer.ts]: Okay, I am going to remove it

Comment (ali-campbell) [reply src/modules/tests/test-data/test-data.reducer.ts]: Yes, if there's no need to know that a fault was removed (e.g. for audit) then we should remove the outcome.  Saves on storage and easier for validation.

**Pull request merged**